colcon_recursive_crawl/package_discovery/recursive_crawl.py

```diff
@@ -44,6 +44,10 @@ def discover(self, *, args, identification_extensions):  # noqa: D102
         if args.base_paths is None:
             return set()
 
+        if isinstance(args.base_paths, str):
+            raise ValueError('Base paths should be a sequence of strings, '
+                             'not a string')
```

Comment:
Imo this check comes way to late and should happen:

1. during the argument parsing phase and
2. the fix should be scoped to `colcon-defaults` since that is the code part which introduces the invalid value

Reply:
How would colcon-defaults know this value is supposed to be a list? I don't think it does. Also, `set_defaults` is part of argparse's configuration, *not* its user interaction model, so it doesn't go through any validation.

I think this is the best I can do, but feel free to explain the approach you recommend.

Reply:
See colcon/colcon-defaults#24 for how this can be achieved within `colcon-defaults` for the majority of arguments including this case.

Based on my previous feedback and the referenced alternative approach I will go ahead and close this.